test/integration/framework/controlplane_utils.go

```diff
@@ -187,6 +187,8 @@ func startAPIServerOrDie(controlPlaneConfig *controlplane.Config, incomingServer
 		}
 		close(stopCh)
 		m.GenericAPIServer.Destroy()
+		// close connections so the server doesn't hang on Close() waiting forever
+		s.CloseClientConnections()
```

Comment:
For my own education: what will happen if the client will try to reestablish the connection after that?

[In other words - isn't that racy in some cases? i.e. if some client wants to send a request, they will try to reestablish the connection if the current one will get closed, no?]

Reply:
good question, it seems the golang people already thought about that and documented it in this comment

https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/net/http/httptest/server.go;l=207-224

as you say, it sounds racy, but the window seems too small :shrug:

Reply:
nit: so that sounds fine to me, but maybe let's add some comment here that we're aware of it and doing that consciously?